morango/syncsession.py

```diff
@@ -70,10 +84,17 @@ class Connection(object):
 
 class NetworkSyncConnection(Connection):
 
-    def __init__(self, base_url=''):
+    def __init__(self, base_url='', compresslevel=9):
         self.base_url = base_url
-
-    def _request(self, endpoint, method="GET", lookup=None, data={}, params={}, userargs=None, password=None):
+        self.compresslevel = compresslevel
+        # ping server at url with info request
+        info_url = urljoin(self.base_url, api_urls.INFO)
+        self.server_info = requests.get(info_url).json()
+        self.capabilities = set()
+        if self.server_info.get('capabilities'):
+            self.capabilities = CAPABILITIES.intersection(set(self.server_info['capabilities']))
+
+    def _request(self, endpoint, method="GET", lookup=None, data={}, params={}, userargs=None, password=None, gzip=False):
```

Comment:
A param called `gzip` suggests to me it's going to do the gzipping inside here, but if I understand it correctly, this is more something like `data_is_gzipped`?